command/resource/client/client.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+const (
+	HeaderConsulToken = "x-consul-token"
+)
+
+type GRPCClient struct {
+	Client pbresource.ResourceServiceClient
+	Config *GRPCConfig
+	Conn   *grpc.ClientConn
+}
+
+func NewGRPCClient(config *GRPCConfig) (*GRPCClient, error) {
+	conn, err := dial(config)
+	if err != nil {
+		return nil, fmt.Errorf("error dialing grpc: %+v", err)
+	}
+	return &GRPCClient{
+		Client: pbresource.NewResourceServiceClient(conn),
+		Config: config,
+		Conn:   conn,
+	}, nil
+}
+
+func (client *GRPCClient) Apply(parsedResource *pbresource.Resource) (*pbresource.Resource, error) {
+	token, err := client.Config.GetToken()
+	if err != nil {
+		return nil, err
+	}
+	ctx := context.Background()
+	if token != "" {
+		ctx = metadata.AppendToOutgoingContext(context.Background(), HeaderConsulToken, token)
+	}
+
+	defer client.Conn.Close()
```

Comment:
Why do we do this logic here? This assumes we can't use the `*GRPCClient` again after calling this method?

Reply:
Yes, we are gonna close the connection because here it's CLI command, and once we finish the call, we turn off the connection.